tests/test_visualization.py

```diff
@@ -1,24 +1,115 @@
+from typing import List  # NOQA
+from typing import Optional  # NOQA
+
 import pytest
 
+from optuna.distributions import CategoricalDistribution
 from optuna.distributions import LogUniformDistribution
 from optuna.distributions import UniformDistribution
+from optuna.structs import StudyDirection
 from optuna.study import create_study
+from optuna.study import Study  # NOQA
 from optuna.trial import Trial  # NOQA
 from optuna import visualization
+from optuna.visualization import _generate_contour_subplot
 from optuna.visualization import _get_contour_plot
 from optuna.visualization import _get_intermediate_plot
 from optuna.visualization import _get_optimization_history_plot
 from optuna.visualization import _get_parallel_coordinate_plot
 from optuna.visualization import _get_slice_plot
 
 
+def _prepare_study_with_trials(no_trials=False, less_than_two=False, with_c_d=True):
+    # type: (bool, bool, bool) -> Study
+    """Prepare a study for tests.
+
+    Args:
+        no_trials: If ``False``, create a study with no trials.
+        less_than_two: If ``True``, create a study with two/four hyperparams where
```

Comment:
Please use `hyperparameters` instead of `hyperparams` as can be seen in the other files (such as https://github.com/pfnet/optuna/blob/master/optuna/integration/lightgbm_tuner/optimize.py#L211 and https://github.com/pfnet/optuna/blob/master/optuna/integration/sklearn.py#L134).

```suggestion
        less_than_two: If ``True``, create a study with two/four hyperparameters where
```